regions/io/ds9/core.py

```diff
@@ -0,0 +1,169 @@
+        ss += '\n'
+        return ss
+
+    def convert_coords(self):
+        """Process list of coordinates
+
+        This mainly seaches for tuple of coordinates in the coordinate list and
+        creates a SkyCoord or PixCoord object from them if appropriate for a
+        given region type. This involves again some coordinate transformation,
+        so this step could be moved to the parsing process
+        """
+        from .read import DS9Parser
+        if self.coordsys in DS9Parser.coordsys_mapping:
+            coords = self._convert_sky_coords()
+        else:
+            coords = self._convert_pix_coords()
+
+        if self.region_type == 'line':
+            coords = [coords[0][0], coords[0][1]]
+
+        return coords
+
+    def _convert_sky_coords(self):
+        """Convert to sky coords"""
+        parsed_angles = [(x, y)
+                         for x, y in zip(self.coord[:-1:2], self.coord[1::2])
+                         if (isinstance(x, coordinates.Angle) and
+                             isinstance(y, coordinates.Angle))
+                        ]
+        frame = coordinates.frame_transform_graph.lookup_name(self.coordsys)
+
+        lon, lat = zip(*parsed_angles)
+        if hasattr(lon, '__len__') and hasattr(lon, '__lat__') and len(lon) == 1 and len(lat==1):
+            # force entries to be scalar if they are length-1
+            lon, lat = u.Quantity(lon[0]), u.Quantity(lat[0])
+        else:
+            # otherwise, they are vector quantitites
+            lon, lat = u.Quantity(lon), u.Quantity(lat)
+        sphcoords = coordinates.UnitSphericalRepresentation(lon, lat)
+        coords = [frame(sphcoords)]
+
+        if self.region_type != 'polygon':
+            coords += self.coord[len(coords * 2):]
+
+        return coords
+
+    def _convert_pix_coords(self):
+        """Convert to pix coords"""
+        if self.region_type in ['polygon', 'line']:
+            # have to special-case polygon in the phys coord case
+            # b/c can't typecheck when iterating as in sky coord case
+            coords = [PixCoord(self.coord[0::2], self.coord[1::2])]
+        else:
+            temp = [_.value for _ in self.coord]
+            coord = PixCoord(temp[0], temp[1])
+            coords = [coord] + temp[2:]
+
+        return coords
+
+    def to_region(self):
+        """Convert to region object
+        """
+        coords = self.convert_coords()
+        log.debug(coords)
+        viz_keywords = ['color', 'dashed', 'width', 'point', 'font']
+
+        if isinstance(coords[0], BaseCoordinateFrame):
+            reg = self.shape_to_sky_region[self.region_type](*coords)
+        elif isinstance(coords[0], PixCoord):
+            reg = self.shape_to_pixel_region[self.region_type](*coords)
+        else:
+            raise DS9RegionParserError("No central coordinate")
+
+        reg.visual = {key: self.meta[key] for key in self.meta.keys() if key in viz_keywords}
```

Comment:
Can we use OrderedDict everywhere instead of dict and preserve order?
In my experience this helps with readability for users, and also testing / debugging for developers.
We can't achieve round-tripping, but order helps a bit, no?